vello/src/lib.rs

```diff
@@ -782,11 +790,17 @@ impl Renderer {
             .as_ref()
             .expect("renderer should have configured surface_format to use on a surface");
         let mut recording = Recording::default();
-        let target_proxy = ImageProxy::new(width, height, ImageFormat::from_wgpu(target.format));
+        let target_proxy = ImageProxy::new(
+            width,
+            height,
+            ImageFormat::from_wgpu(target.format)
+                .expect("`TargetTexture` always has a supported texture format"),
+        );
```

Comment:
Just checking, but why error here and return an Error below? No value in having a Result from here as well?

And same for above in the same code structure ...

Reply:
If you follow the data flow, it's impossible for this to fail. Returning an error would suggest it's possible for the user to address a failure here, which it wouldn't be.

The whole design of the types here is quite complicated, so I just did the minimum thing which worked.

Reply:
SGTM